src/store/store.ts

```diff
@@ -0,0 +1,1448 @@
+      }
+      if (this.isSpecialResolutionFiling) return FilingNames.SPECIAL_RESOLUTION
+      return null
+    },
+
+    /** The completing party data. */
+    getCompletingParty (): CompletingPartyIF {
+      return this.stateModel.completingParty
+    },
+
+    /** The special resolution object. */
+    getSpecialResolution (): SpecialResolutionIF {
+      return this.stateModel.specialResolution
+    },
+
+    /** The special resolution validity flags. */
+    getSpecialResolutionFormValid (): boolean {
+      return this.getValidationFlags.flagsCompanyInfo.isValidCreateSpecialResolution
+    },
+
+    /** The company info page validity flags. */
+    getSpecialResolutionConfirmValid (): boolean {
+      return this.getFlagsReviewCertify.isValidSpecialResolutionConfirm
+    },
+
+    /** The restoration object. */
+    getRestoration (): RestorationStateIF {
+      return this.stateModel.restoration
+    },
+
+    /** The restoration object. */
+    getStateFilingRestoration (): StateFilingRestorationIF {
+      return this.stateModel.stateFilingRestoration
+    },
+
+    /** The approval type validity. */
+    getApprovalTypeValid (): boolean {
+      return this.stateModel.validationFlags.flagsCompanyInfo.isValidApprovalType
+    },
+
+    /** The expiry date validity. */
+    getExpiryValid (): boolean {
+      return this.stateModel.validationFlags.flagsCompanyInfo.isValidExtensionTime
+    },
+
+    /** Returns false when users can change the sole proprietor (SP).
+     * Restricts the ability of non-staff users from changing the sole
+     * proprietor when the SP is an organization.  This restriction has been
+     * added to ensure that changes made in business-edit-ui are also updated by
+     * staff in COLIN */
+    hideChangeButtonForSoleProps (): boolean {
+      const isProprietor = this.getOrgPeople[0]?.roles[0]?.roleType === RoleTypes.PROPRIETOR
+      const isOrganization = this.getOrgPeople[0]?.officer?.partyType === PartyTypes.ORGANIZATION
+      const isDba = isProprietor && isOrganization
+      return !this.isRoleStaff && isDba
+    },
+
+    getExpiryDateString (): string {
+      return this.stateModel.restoration?.expiry
+    },
+
+    getFormattedExpiryText: (state) => (today = new Date()): string => {
+      if (state.stateModel.restoration?.expiry) {
+        const numberOfExtensionMonths = DateUtilities.subtractDates(state.stateModel.stateFilingRestoration?.expiry,
+          state.stateModel.restoration?.expiry)
+        const expiryDatePacific = DateUtilities.yyyyMmDdToPacificDate(state.stateModel.restoration?.expiry)
+        const formattedExpiryText = numberOfExtensionMonths + ' months, expires on ' + expiryDatePacific
+        return formattedExpiryText
+      }
+      return '[no expiry date]'
+    }
+  },
+  actions: {
+    // no context as first argument, use `this` instead
+    setEntityType (entityType: CorpTypeCd) {
+      this.stateModel.tombstone.entityType = entityType
```

Comment:
So actions are the new mutations?

Reply:
Yes

Reply:
Might be a good idea, not sure if the class way can identify through typescript if the action returns a promise or not

Reply:
https://pinia.vuejs.org/cookbook/options-api.html

Reply:
Doesn't look as explicit as `@Action` though

Reply:
1. It remains to be seen whether `mapStores` passes along the typings.

2. I hope we never use `mapWritableState` and always use actions instead. I'm thinking the actual store state should always use accessors as a best practise.